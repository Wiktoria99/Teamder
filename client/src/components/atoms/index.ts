export * from './BackButton';
export * from './CustomTextField';
export * from './CustomButton';
export * from './SideBarButton';
