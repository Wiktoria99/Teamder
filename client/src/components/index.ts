export * from "./atoms";
export * from "./atoms";
export * from "./molecules";
export * from "./organisms";
