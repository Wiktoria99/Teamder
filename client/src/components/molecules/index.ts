export * from './RegisterSteps';
export * from './TeamItem';
export * from './Suggestions';
export * from './Sidebar';
export * from './TeamList';
export * from './SuggestionItem';
export * from './SuggestionList';
export * from './NotificationItem';
export * from './NotificationList';
