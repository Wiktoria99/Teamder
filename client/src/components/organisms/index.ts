export * from './Layout';
export * from './FormWrapper';
export * from './MainWrapper';
export * from './BarWrapper';
