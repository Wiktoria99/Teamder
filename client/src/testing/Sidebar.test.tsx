import ReactDOM from 'react-dom';
import { BrowserRouter as Router, NavLink } from 'react-router-dom';
import { StaticRouter } from 'react-router';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';
import { Sidebar } from '@/components';
import { cleanup } from '@testing-library/react';

afterEach(cleanup);

it('Sidebar renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(
    <Router>
      <Sidebar />
    </Router>,
    div,
  );
});

it('Sidebar matches snapshot', () => {
  const tree = renderer
    .create(
      <Router>
        <Sidebar />
      </Router>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('NavLink matches snapshot', () => {
  const context = {};
  const tree = renderer
    .create(
      <StaticRouter location="loc" context={context}>
        <NavLink to="#" />
      </StaticRouter>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('NavLink to my_profile matches snapshot', () => {
  const context = {};
  const tree = renderer
    .create(
      <StaticRouter location="loc" context={context}>
        <NavLink to="/my_profile" />
      </StaticRouter>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('NavLink to mainpage matches snapshot', () => {
  const context = {};
  const tree = renderer
    .create(
      <StaticRouter location="loc" context={context}>
        <NavLink to="/" />
      </StaticRouter>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('NavLink to my_teams matches snapshot', () => {
  const context = {};
  const tree = renderer
    .create(
      <StaticRouter location="loc" context={context}>
        <NavLink to="/my_teams" />
      </StaticRouter>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('NavLink to notifications matches snapshot', () => {
  const context = {};
  const tree = renderer
    .create(
      <StaticRouter location="loc" context={context}>
        <NavLink to="/notifications" />
      </StaticRouter>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
