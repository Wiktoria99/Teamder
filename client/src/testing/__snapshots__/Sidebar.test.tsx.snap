// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavLink matches snapshot 1`] = `
<a
  aria-current="page"
  className="active"
  href="/loc"
  onClick={[Function]}
  style={Object {}}
/>
`;

exports[`NavLink to mainpage matches snapshot 1`] = `
<a
  aria-current={null}
  href="/"
  onClick={[Function]}
/>
`;

exports[`NavLink to my_profile matches snapshot 1`] = `
<a
  aria-current={null}
  href="/my_profile"
  onClick={[Function]}
/>
`;

exports[`NavLink to my_teams matches snapshot 1`] = `
<a
  aria-current={null}
  href="/my_teams"
  onClick={[Function]}
/>
`;

exports[`NavLink to notifications matches snapshot 1`] = `
<a
  aria-current={null}
  href="/notifications"
  onClick={[Function]}
/>
`;

exports[`Sidebar matches snapshot 1`] = `
<div
  className="MuiBox-root MuiBox-root-12 makeStyles-barContainer-5"
>
  <div
    className="MuiBox-root MuiBox-root-13 makeStyles-barBox-6"
  >
    <div
      className="MuiBox-root MuiBox-root-14 makeStyles-mainContainer-1"
    >
      <svg
        className="makeStyles-svg-10"
        fill="none"
        height="32"
        onClick={[Function]}
        viewBox="0 0 172 32"
        width="172"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M25.3501 15.5327C25.3453 14.0653 25.1794 12.6023 24.855 11.1673H29C28.7865 10.5688 28.5288 9.9853 28.2286 9.42118H24.3714C23.6463 7.12504 22.5217 4.95984 21.0439 3.01496C20.0998 2.63866 19.1153 2.36056 18.1079 2.18553C20.1324 4.24735 21.6606 6.70119 22.5868 9.37753H16.3808V2H14.6537V9.38844H8.44776C9.37586 6.70726 10.9081 4.24949 12.9382 2.18553C11.935 2.35764 10.9545 2.63207 10.0136 3.00404C8.53003 4.94078 7.39765 7.09852 6.66311 9.38844H2.78294C2.47733 9.96286 2.21567 10.5574 2 11.1673H6.17953C5.85517 12.6023 5.68925 14.0653 5.68444 15.5327C5.68762 17.1379 5.88072 18.7376 6.26013 20.3019H2.18422C2.42242 20.9137 2.70718 21.5082 3.03625 22.0808H6.75522C7.46011 24.1716 8.50511 26.1453 9.85245 27.9305C10.8173 28.3227 11.8252 28.6119 12.8576 28.7926C10.9784 26.8498 9.53324 24.5667 8.60896 22.0808H14.6652V29H16.3923V22.0808H22.4256C21.4982 24.5678 20.049 26.8509 18.1655 28.7926C19.2027 28.6057 20.2145 28.3091 21.1821 27.9087C22.5273 26.1302 23.5722 24.164 24.2793 22.0808H27.9638C28.2913 21.5188 28.5761 20.9352 28.8158 20.3347H24.7284C25.1258 18.7614 25.3344 17.1505 25.3501 15.5327ZM14.6537 20.3019H8.04478C7.23417 17.2984 7.19862 14.1539 7.94115 11.1346H14.6537V20.3019ZM22.9898 20.3019H16.3808V11.1673H23.0934C23.4241 12.6011 23.5863 14.0651 23.577 15.5327C23.5863 17.1392 23.3891 18.7407 22.9898 20.3019Z"
          fill="white"
        />
        <path
          d="M15.5556 0C12.479 0 9.47146 0.912317 6.91336 2.62158C4.35526 4.33085 2.36147 6.7603 1.1841 9.6027C0.00673993 12.4451 -0.301312 15.5728 0.298903 18.5903C0.899117 21.6078 2.38064 24.3795 4.55612 26.555C6.73161 28.7305 9.50334 30.212 12.5208 30.8122C15.5383 31.4124 18.666 31.1044 21.5084 29.927C24.3508 28.7497 26.7803 26.7559 28.4895 24.1978C30.1988 21.6397 31.1111 18.6322 31.1111 15.5556C31.1111 11.43 29.4722 7.47335 26.555 4.55612C23.6378 1.63888 19.6812 0 15.5556 0ZM15.5556 29.1667C12.8635 29.1667 10.232 28.3684 7.99363 26.8728C5.7553 25.3772 4.01073 23.2514 2.98054 20.7643C1.95034 18.2772 1.6808 15.5405 2.20599 12.9002C2.73117 10.2599 4.02751 7.83459 5.93105 5.93105C7.8346 4.0275 10.2599 2.73117 12.9002 2.20598C15.5405 1.68079 18.2772 1.95034 20.7643 2.98053C23.2514 4.01072 25.3772 5.75529 26.8728 7.99363C28.3684 10.232 29.1667 12.8635 29.1667 15.5556C29.1667 19.1654 27.7327 22.6275 25.1801 25.1801C22.6275 27.7326 19.1655 29.1667 15.5556 29.1667Z"
          fill="white"
        />
        <path
          d="M59.3545 8.6416H52.9531V26H47.8115V8.6416H41.5273V4.67188H59.3545V8.6416ZM73.4146 16.9619H65.3287V22.0449H74.8795V26H60.1871V4.67188H74.9088V8.6416H65.3287V13.1387H73.4146V16.9619ZM87.7385 22.0156H80.6926L79.4621 26H73.9689L81.7912 4.67188H86.6252L94.5061 26H88.9836L87.7385 22.0156ZM81.923 18.0459H86.508L84.2082 10.6484L81.923 18.0459ZM101.476 4.67188L106.237 19.5107L110.983 4.67188H117.751V26H112.595V21.0195L113.093 10.8242L107.936 26H104.538L99.367 10.8096L99.865 21.0195V26H94.7234V4.67188H101.476ZM119.682 26V4.67188H126.552C128.437 4.67188 130.131 5.10156 131.635 5.96094C133.139 6.81055 134.311 8.0166 135.151 9.5791C136 11.1318 136.43 12.875 136.44 14.8086V15.79C136.44 17.7432 136.025 19.4961 135.195 21.0488C134.374 22.5918 133.212 23.8027 131.708 24.6816C130.214 25.5508 128.544 25.9902 126.699 26H119.682ZM124.824 8.6416V22.0449H126.611C128.085 22.0449 129.218 21.5225 130.009 20.4775C130.8 19.4229 131.196 17.8604 131.196 15.79V14.8672C131.196 12.8066 130.8 11.2539 130.009 10.209C129.218 9.16406 128.066 8.6416 126.552 8.6416H124.824ZM150.954 16.9619H142.868V22.0449H152.419V26H137.727V4.67188H152.448V8.6416H142.868V13.1387H150.954V16.9619ZM161.088 18.4707H158.305V26H153.164V4.67188H161.557C164.087 4.67188 166.064 5.2334 167.49 6.35645C168.916 7.47949 169.629 9.06641 169.629 11.1172C169.629 12.6016 169.326 13.832 168.72 14.8086C168.125 15.7852 167.187 16.5762 165.908 17.1816L170.361 25.7803V26H164.853L161.088 18.4707ZM158.305 14.501H161.557C162.534 14.501 163.266 14.2471 163.754 13.7393C164.253 13.2217 164.502 12.5039 164.502 11.5859C164.502 10.668 164.253 9.9502 163.754 9.43262C163.256 8.90527 162.524 8.6416 161.557 8.6416H158.305V14.501Z"
          fill="white"
        />
      </svg>
      <div
        className="makeStyles-routeList-3"
        style={
          Object {
            "marginTop": "45px",
          }
        }
      >
        <a
          aria-current="page"
          className="sc-bdvvaa fyBRYB active"
          href="/"
          onClick={[Function]}
          style={Object {}}
        >
          <div
            className="MuiGrid-root makeStyles-linkItem-4 MuiGrid-container MuiGrid-align-items-xs-flex-end"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 5.69l5 4.5V18h-2v-6H9v6H7v-7.81l5-4.5M12 3L2 12h3v8h6v-6h2v6h6v-8h3L12 3z"
              />
            </svg>
            <span
              className="sc-gsDJrp gVGABg"
            >
              Strona główna
            </span>
          </div>
        </a>
        <a
          aria-current={null}
          className="sc-bdvvaa fyBRYB"
          href="/my_teams"
          onClick={[Function]}
        >
          <div
            className="MuiGrid-root makeStyles-linkItem-4 MuiGrid-container MuiGrid-align-items-xs-flex-end"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M9 13.75c-2.34 0-7 1.17-7 3.5V19h14v-1.75c0-2.33-4.66-3.5-7-3.5zM4.34 17c.84-.58 2.87-1.25 4.66-1.25s3.82.67 4.66 1.25H4.34zM9 12c1.93 0 3.5-1.57 3.5-3.5S10.93 5 9 5 5.5 6.57 5.5 8.5 7.07 12 9 12zm0-5c.83 0 1.5.67 1.5 1.5S9.83 10 9 10s-1.5-.67-1.5-1.5S8.17 7 9 7zm7.04 6.81c1.16.84 1.96 1.96 1.96 3.44V19h4v-1.75c0-2.02-3.5-3.17-5.96-3.44zM15 12c1.93 0 3.5-1.57 3.5-3.5S16.93 5 15 5c-.54 0-1.04.13-1.5.35.63.89 1 1.98 1 3.15s-.37 2.26-1 3.15c.46.22.96.35 1.5.35z"
              />
            </svg>
            <span
              className="sc-gsDJrp gVGABg"
            >
              Moje zespoły
            </span>
          </div>
        </a>
        <a
          aria-current={null}
          className="sc-bdvvaa fyBRYB"
          href="/notifications"
          onClick={[Function]}
        >
          <div
            className="MuiGrid-root makeStyles-linkItem-4 MuiGrid-container MuiGrid-align-items-xs-flex-end"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M22 6c0-1.1-.9-2-2-2H4c-1.1 0-2 .9-2 2v12c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6zm-2 0l-8 5-8-5h16zm0 12H4V8l8 5 8-5v10z"
              />
            </svg>
            <span
              className="sc-gsDJrp gVGABg"
            >
              Powiadomienia
            </span>
          </div>
        </a>
        <a
          aria-current={null}
          className="sc-bdvvaa fyBRYB"
          href="/my_profile"
          onClick={[Function]}
        >
          <div
            className="MuiGrid-root makeStyles-linkItem-4 MuiGrid-container MuiGrid-align-items-xs-flex-end"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 6c1.1 0 2 .9 2 2s-.9 2-2 2-2-.9-2-2 .9-2 2-2m0 10c2.7 0 5.8 1.29 6 2H6c.23-.72 3.31-2 6-2m0-12C9.79 4 8 5.79 8 8s1.79 4 4 4 4-1.79 4-4-1.79-4-4-4zm0 10c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
              />
            </svg>
            <span
              className="sc-gsDJrp gVGABg"
            >
              Profil
            </span>
          </div>
        </a>
      </div>
      <a
        aria-current={null}
        href="/create_team"
        onClick={[Function]}
        style={
          Object {
            "textDecoration": "none",
          }
        }
      >
        <button
          className="MuiButtonBase-root MuiButton-root WithStyles(ForwardRef(Button))-root-11 MuiButton-text makeStyles-button-2 MuiButton-textSecondary"
          disabled={false}
          onBlur={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="submit"
        >
          <span
            className="MuiButton-label"
          >
            Stwórz zespół
          </span>
        </button>
      </a>
    </div>
    <button
      className="MuiButtonBase-root MuiButton-root WithStyles(ForwardRef(Button))-root-11 MuiButton-text makeStyles-button-2 MuiButton-textSecondary"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="submit"
    >
      <span
        className="MuiButton-label"
      >
        WYLOGUJ
      </span>
    </button>
  </div>
</div>
`;
